### PR TITLE
Refactored existing R1 Tests

### PR DESCRIPTION
**Pull request by eileenvanheerde**
Refactored all the R1 tests so that they actually do stuff.

### CODE REVIEW

File: src/__tests__/ParserComponent.test.tsx [hunk 1]
```diff
@@ -36,51 +38,126 @@ describe('<ParserComponent /> renders correctly', () => {
...
-    comp.find('input').simulate('change', event);
-    const fileUsed: File = onChangeMock.mock.calls[0][0];
-    expect(fileUsed.name).toBe(testFile.name);
-    expect(onChangeMock).toHaveBeenCalledTimes(1);
-  });
+  // store original console.warn
+  const originalWarn = console.warn;
+
+  // array to store console output
+  const consoleOutput: any[] = [];
```
Comment (braunsonm) [inline src/__tests__/ParserComponent.test.tsx]: WHy?

File: src/components/ParserComponent.tsx [hunk 2]
```diff
@@ -16,354 +16,354 @@ import 'bootstrap/dist/css/bootstrap.min.css';
  */
 export default class ParserComponent extends React.Component<ParserInterface,
   ParserState> {
-    private columnTypes?: Array<Column>;
```
Comment (braunsonm) [inline src/components/ParserComponent.tsx]: Why did you reformat this entire file? I'd revert this change.

File: src/components/TimelineComponent.tsx [hunk 3]
```diff
@@ -499,6 +500,17 @@ export default class TimelineComponent
         .style('color', 'red')
         .text('yColumn');
 
+    d3.selectAll('.tick').each(function(this: any, d: any, i: number) {
+      // eslint-disable-next-line no-invalid-this
+      const transform = d3.select(this).attr('transform');
+      const newX =
+          Number.parseFloat(transform.split(',')[1].split(')')[0]).toFixed(4);
+
+      console.log(newX);
```
Comment (braunsonm) [inline src/components/TimelineComponent.tsx]: Remove

File: src/components/TimelineComponent.tsx [hunk 4]
```diff
@@ -520,14 +532,15 @@ export default class TimelineComponent
...
+                `translate(${(m.width / 2).toFixed(4)},
+                ${(m.height + m.marginTop + 20).toFixed(4)})`)
             .style('text-anchor', 'start')
             .text(this.state.xColumn);
 
         this.svg.append('text')
             .attr('transform', 'rotate(-90)')
-            .attr('y', 0 - m.marginLeft)
-            .attr('x', 0 - (m.height / 2))
+            .attr('y', (0 - m.marginLeft).toFixed(4))
```
Comment (braunsonm) [inline src/components/TimelineComponent.tsx]: Can you clarify why you fixed these numbers?

Comment (mkoma808) [reply src/components/TimelineComponent.tsx]: this was done because the snapshot tests were failing because of incredibly small floating point differences that only occurred when testing on sonarcloud.

File: src/components/TimelineComponent.tsx (hunk 3 above)
Comment (eileenvanheerde) [reply src/components/TimelineComponent.tsx]: i merged develop, so it's gone now

File: src/__tests__/ParserComponent.test.tsx (hunk 1 above)
Comment (eileenvanheerde) [reply src/__tests__/ParserComponent.test.tsx]: forgot to remove this. it was just to make the console a little easier to read because momentjs kept printing a warning about a deprecated function (which should probably be addressed).
image

File: src/components/ParserComponent.tsx (hunk 2 above)
Comment (eileenvanheerde) [reply src/components/ParserComponent.tsx]: Sorry, I had a bit of a smooth brain moment and instead of reverting it i just copy & pasted develop's ParserComponent.tsx 😅 Forgive me 🙏

File: src/__tests__/ParserComponent.test.tsx (hunk 1 above)
Comment (eileenvanheerde) [reply src/__tests__/ParserComponent.test.tsx]: left it in, but commented out.
